docs/install/installing-istio.md

```diff
@@ -0,0 +1,280 @@
+---
+title: "Installing Istio for Knative"
+weight: 15
+type: "docs"
+---
+
+This guide walks you through manually installing and customizing Istio for use
+with Knative.
+
+If your cloud platform offers a managed Istio installation, we recommend
+installing Istio that way, unless you need the ability to customize your
+installation. If your cloud platform offers a managed Istio installation,
+the [install guide](./README.md) for your specific platform will have those instructions.
+
+For example, the [GKE Install Guide](./knative-with-gke.md) includes the
+instructions for installing Istio on your cluster using `gcloud`.
+
+## Before you begin
+
+You need:
+- A Kubernetes cluster created.
+- [`helm`](https://helm.sh/) installed.
+
+## Installing Istio
+
+When you install Istio, there are a few options depending on your goals. For a
+basic Istio installation suitable for most Knative use cases, follow the
+[Installing Istio with sidecar injection](#installing-istio-with-sidecar-injection)
+instructions. If you're familiar with Istio and know what kind of
+installation you want, read through the options and choose the installation that
+suits your needs.
+
+You can easily customize your Istio installation with `helm`. The below sections
+cover a few useful Istio configurations and their benefits. 
+
+### Choosing an Istio installation
+
+You can install Istio with or without a service mesh:
+
+- _automatic sidecar injection_: Enables the Istio service mesh by
+  [automatically injecting the Istio sidecars][1]. The sidecars are injected
+  into each pod of your cluster as they are created.
+
+- _manual sidecar injection_: Provides your Knative installation with traffic
+  routing and ingress, without the Istio service mesh. You do have the option of
+  later enabling the service mesh if you [manually inject the Istio sidecars][2].
+
+If you are just getting started with Knative, you should choose automatic
+sidecar injection and enable the Istio service mesh.
+
+Due to current dependencies, some installable Knative options require the Istio
```

Comment:
Are lines 51-53 still true? If they are, I would lean towards installing Istio with automatic sidecar injection as the default, since the majority of the install instructions right now tell people to install all the components as well as the observability plug-ins. I don't want to recommend a default that doesn't work for a lot of use cases...

Reply:
@ZhiminXiang  @mattmoor

Reply:
+ @grantr 
I guess in eventing side we are working on getting rid of mesh. I think Grant probably knows better about this part.

Reply:
Eventing 0.6 no longer requires Istio. :tada: @akashrv worked on this, so he knows the most about it.

Reply:
That is correct. Eventing doesn't require Istio, but would continue to work with Istio mesh.